presto-session-property-managers/src/main/java/io/prestosql/plugin/session/SessionMatchSpec.java

```diff
@@ -51,42 +52,47 @@ public SessionMatchSpec(
             @JsonProperty("clientTags") Optional<List<String>> clientTags,
             @JsonProperty("queryType") Optional<String> queryType,
             @JsonProperty("group") Optional<Pattern> resourceGroupRegex,
-            @JsonProperty("sessionProperties") Map<String, String> sessionProperties)
+            @JsonProperty("sessionProperties") Map<String, String> sessionProperties,
+            @JsonProperty("catalogSessionProperties") Map<String, Map<String, String>> catalogSessionProperties)
     {
         this.userRegex = requireNonNull(userRegex, "userRegex is null");
         this.sourceRegex = requireNonNull(sourceRegex, "sourceRegex is null");
         requireNonNull(clientTags, "clientTags is null");
         this.clientTags = ImmutableSet.copyOf(clientTags.orElse(ImmutableList.of()));
         this.queryType = requireNonNull(queryType, "queryType is null");
         this.resourceGroupRegex = requireNonNull(resourceGroupRegex, "resourceGroupRegex is null");
-        requireNonNull(sessionProperties, "sessionProperties is null");
-        this.sessionProperties = ImmutableMap.copyOf(sessionProperties);
+        if (sessionProperties == null && catalogSessionProperties == null) {
```

Comment:
Adding 

`@JsonProperty("catalogSessionProperties") Map<String, Map<String, String>> catalogSessionProperties)`

Cause the configuration to look like:
```
  {
    "group": "global.*",
    "sessionProperties": {
      "query_max_execution_time": "8h",
    }
   "catalogSessionProperties": {
       "hive": {
         "insert_existing_partitions_behavior": "overwrite"
       }
     }
  },
```

Alternative implementation can be to not change the Json stucture :
```
  {
    "group": "global.*",
    "sessionProperties": {
      "query_max_execution_time": "8h",
       "hive.insert_existing_partitions_behavior": "overwrite"
     }
  },
```

and handle the distinction between system & catalog properties here with something that look like:

```
        requireNonNull(sessionProperties, "sessionProperties is null");
        Map<String, String> systemSessionProperties = new HashMap<>();
        Map<String, Map<String, String>> catalogsSessionProperties = new HashMap<>();
        sessionProperties.entrySet()
                .stream()
                .forEach(sessionPropertyEntry -> {
                    String[] property = sessionPropertyEntry.getKey().split("\\.");
                    switch (property.length) {
                        case 1:
                            systemSessionProperties.put(sessionPropertyEntry.getKey(), sessionPropertyEntry.getValue());
                        case 2: 
                            catalogsSessionProperties.compute(property[0], (catalog, properties) -> {
                                Map<String, String> catalogProperties = Optional.ofNullable(properties).orElseGet(HashMap::new);
                                catalogProperties.put(sessionPropertyEntry.getKey(), sessionPropertyEntry.getValue());
                                return catalogProperties;
                            });
                        default: // throw new Exception("TODO presto config exception");
                    }});
        catalogsSessionProperties.replaceAll((key, val) -> ImmutableMap.copyOf(val));
        this.sessionProperties = new SessionProperties(
                ImmutableMap.copyOf(sessionProperties),
                ImmutableMap.copyOf(catalogsSessionProperties));
```